sandy/core/cov.py

```diff
@@ -698,7 +740,7 @@ def gls_update(self,  S, Vy, threshold=None):
         """
         index, columns = self.data.index, self.data.columns
         Vx = self.data.values
-        A = self._gls_sensitivity(S, Vy, threshold).values
+        A = self._gls_var_sensitivity(S, Vy, threshold).values
         V_new = Vx - A.dot(Vx)
         V_new = pd.DataFrame(V_new, index=index, columns=columns)
         return self.__class__(V_new)

```

Comment:
you can just do `return self.__class__(V_new, index=index, columns=columns)` without the previous line

Reply:
Give me an error if i put in that way

Reply:
strange...